transport/src/main/java/io/netty/channel/SliceableFileRegion.java

```diff
@@ -0,0 +1,131 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+import java.io.IOException;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ * A {@link FileRegion} which supports slice.
+ * <p>
+ * We introduce this interface because we can not add new methods to an interface until next
+ * minor release. All the methods here will be moved to FileRegion when we release netty 4.2.
+ */
+public interface SliceableFileRegion extends FileRegion {
+
+    /**
+     * Returns the {@code transferIndex} of this file region.
+     */
+    long transferIndex();
+
+    /**
+     * Sets the {@code transferIndex} of file region.
+     *
+     * @throws IndexOutOfBoundsException
+     *             if the specified {@code transferIndex} is less than {@code 0} or greater than
+     *             {@link #count()}
+     */
+    SliceableFileRegion transferIndex(long index);
+
+    /**
+     * Returns {@code true} if and only if {@code (this.count - this.transferIndex)} is greater than
+     * {@code 0}.
+     */
+    boolean isTransferable();
+
+    /**
+     * Returns the number of readable bytes which is equal to {@code (count - transferIndex)}.
+     */
+    long transferableBytes();
+
+    /**
+     * Transfers the content of this file region to the specified channel.
+     *
+     * @param target
+     *            the destination of the transfer
+     * @param length
+     *            the maximum number of bytes to transfer
+     */
+    long transferBytesTo(WritableByteChannel target, long length) throws IOException;
+
+    /**
+     * Transfers the content of this file region to the specified channel.
+     * <p>
+     * This method does not modify {@link #transferIndex()}.
+     *
+     * @param target
+     *            the destination of the transfer
+     * @param position
+     *            the relative offset of the file where the transfer begins from. For example,
+     *            <tt>0</tt> will make the transfer start from {@link #position()}th byte and
+     *            <tt>{@link #count()} - 1</tt> will make the last byte of the region transferred.
+     * @param length
+     *            the maximum number of bytes to transfer
+     */
+    long transferBytesTo(WritableByteChannel target, long position, long length) throws IOException;
+
+    /**
+     * Returns a slice of this file region's sub-region. This method does not modify
+     * {@code transferIndex} of this buffer.
+     * <p>
+     * Also be aware that this method will NOT call {@link #retain()} and so the reference count
+     * will NOT be increased.
+     */
+    SliceableFileRegion slice(long index, long length);
+
+    /**
+     * Returns a slice of this file region's sub-region. This method does not modify
+     * {@code transferIndex} of this buffer.
+     * <p>
+     * Note that this method returns a {@linkplain #retain() retained} file region unlike
+     * {@link #slice()}.
```

Comment:
this link is dead -> `{@link #slice(long, long)}`

Reply:
Done.